arelle/plugin/validate/DBA/rules/fr.py

```diff
@@ -71,6 +72,48 @@ def rule_fr7(
     )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_fr34(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    DBA.FR34: If Equity does not equal 0 more fields are required.  At least one field must be filled in on the balance sheet in addition to equity:
+    Assets, NoncurrentAssets, CurrentAssets, LongtermLiabilitiesOtherThanProvisions, ShorttermLiabilitiesOtherThanProvisions, LiabilitiesOtherThanProvisions, LiabilitiesAndEquity
+    """
+    equityFacts = val.modelXbrl.factsByQname.get(pluginData.equityQn)
+    foundEquityFact = False
+    otherRequiredFacts = []
+    nonZeroEquityFacts = []
+    if equityFacts:
+        for fact in equityFacts:
+            if fact.xValid >=VALID:
+                if fact.xValue != 0:
+                    nonZeroEquityFacts.append(fact)
+                    foundEquityFact = True
+    if foundEquityFact:
+        otherRequiredFactsQnames = [
+            pluginData.assetsQn, pluginData.noncurrentAssetsQn, pluginData.longtermLiabilitiesOtherThanProvisionsQn,
+            pluginData.shorttermLiabilitiesOtherThanProvisionsQn, pluginData.liabilitiesOtherThanProvisionsQn, pluginData.liabilitiesAndEquityQn
+        ]
+        for factQname in otherRequiredFactsQnames:
+            facts = val.modelXbrl.factsByQname.get(factQname)
+            if facts:
+                otherRequiredFacts.append(facts)
```

Comment:
It doesn't look like we actually need to collect these facts, we just need to know if any of them exist and can break as soon as we find one.

```python
hasEquityRequiredFacts = False
for factQname in otherRequiredFactsQnames:
    facts = val.modelXbrl.factsByQname.get(factQname)
    if facts:
        hasEquityRequiredFacts = True
        break
```
or:
```python
hasEquityRequiredFacts = any(val.modelXbrl.factsByQname.get(factQname) for factQname in otherRequiredFactsQnames)
```